Caboodle/Permissions/Permissions.ios.cs

```diff
@@ -9,6 +9,10 @@ namespace Microsoft.Caboodle
 {
     internal static partial class Permissions
     {
+        static readonly object locationLocker = new object();
+        static CLLocationManager locationManager;
+        static TaskCompletionSource<PermissionStatus> locationTcs;
```

Comment:
This needs a set of eyes here 👀

Reply:
I am locking and making sure that there is only a single location manager for all permissions, but is this necessary?

Reply:
I also saw a constructor for `TaskCompletionSource` that takes `object state`... I have never used that, but maybe there is some way to use that to keep the listener alive... will investigate, but comments will help.